### PR TITLE
feat(Fossid-webapp): Increase the read timeout for `listMatchedLines`

### PR DESCRIPTION
**Pull request by nnobelis**
#
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt [hunk 1]
```diff
@@ -254,6 +254,7 @@ interface FossIdRestService {
     suspend fun listSnippets(@Body body: PostRequestBody): PolymorphicResponseBody<Snippet>
```
Comment (oheger-bosch) [inline clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:263]: Commit message typo: "shis commit" -> "this commit"

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt [hunk 2]
```diff
@@ -41,6 +41,8 @@ import okhttp3.Interceptor
 import okhttp3.OkHttpClient
```
Comment (oheger-bosch) [inline clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:41]: Commit message: "When the request interceptor *runs* in a timeout ..."

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt [hunk 3]
```diff
@@ -209,7 +211,14 @@ interface FossIdRestService {
 
             val readTimeout = newReadTimeout?.toIntOrNull() ?: chain.readTimeoutMillis()
 
-            return chain.withReadTimeout(readTimeout, TimeUnit.MILLISECONDS).proceed(request)
+            return runCatching {
+                chain.withReadTimeout(readTimeout, TimeUnit.MILLISECONDS).proceed(request)
+            }.onFailure {
+                logger.error { "Error when intercepting the call chain: ${it.message}" }
+                val requestBuffer = Buffer()
+                request.body?.writeTo(requestBuffer)
```
Comment (oheger-bosch) [inline clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:219]: I assume, request bodies cannot contain sensitive information, right? So it should be safe to log them. Do all requests have a body?

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt (hunk 2 above)
Comment (nnobelis) [reply clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:41]: Commit has been removed.

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt (hunk 3 above)
Comment (nnobelis) [reply clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:219]: Argh you are right: FossID request bodies contain the api key :(
I removed this commit as I don't see an elegant workaround.
Maybe the `HttpLoggingInterceptor` of OkHttp: https://stackoverflow.com/a/56892146.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**